app/controllers/concerns/sufia/file_sets_controller_behavior.rb

```diff
@@ -98,12 +98,23 @@ def file_set_params
       end
 
       def find_parent_by_id
-        return if parent_id.empty?
+        return default_work unless parent_id.present?
         super
       end
 
+      # If the user is creating a bunch of files, and not in a work context,
+      # create a work for each file.
+      def default_work
+        # Ensure the upload set exists, before trying to associate a work with it.
+        UploadSet.find_or_create(params[:upload_set_id])
+        GenericWork.create!(title: [params[:file_set][:files].first.original_filename],
+                            upload_set_id: params[:upload_set_id]) do |w|
```

Comment:
could you pass the UploadSet directly to create! since you'd already have it in-hand in the line above? It would eliminate an extra Fedora call, maybe.